File: python/series.py
#!/usr/bin/python3
import math
import time

N = int(input("Insert N: "))
an = 0.0
an_prev = 0.0

with open("series.dat", "w") as file:
    file.write("n\tΣ(1/n^2)\n")

    tic = time.time()
    
    for n in range(1, N + 1):
        an += math.pow(n, -2)
        file.write(f"{n}\t{an:.9f}\n")
        
        if n > 1 and abs(an_prev - an) <= math.pow(10, -9):
            break
        
        an_prev = an
    
    toc = time.time()
    duration = toc - tic
    Time = duration * 1000
    
    print(f"Tempo impiegato: %i ms"%Time)

File: C++/series.cpp
#include <iostream>
#include <string>
#include <iomanip>
#include <fstream>
#include <cmath>
#include <chrono>
using namespace std;

int main(){
	int N;
	float an = 0;
	float an_prev, Time;
    cout << "Insert N: ";
    cin >> N;
    ofstream file("series.dat");
    file << "n" << "\t" << "Σ(1/n^2)" << endl;   
    chrono::steady_clock::time_point tic = chrono::steady_clock::now();
    int n;
    for (n = 1; n <= N; n++){
    	an += pow(n,-2);
    	file << n << "\t" << fixed << setprecision(9) << an << endl;
    	if (n > 1 and abs(an_prev - an) <= pow(10,-6)){
    		break;
    	}
    	an_prev = an;
    }
    file.close();
    chrono::steady_clock::time_point toc = chrono::steady_clock::now();
    chrono::milliseconds duration = chrono::duration_cast<chrono::milliseconds>(toc - tic);
    Time = duration.count();
    cout << "Tempo impiegato: " << Time << " ms" << endl;
	return 0;
}